src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -144,7 +144,8 @@ def _get_mgmt_service_client(cli_ctx,
     else:
         client = client_type(cred, **client_kwargs)
 
-    configure_common_settings(cli_ctx, client)
+    if hasattr(client, 'config'):
+        configure_common_settings(cli_ctx, client)
```

Comment:
You probably want a version of `configure_common_settings`, that will build a set of kwargs to pass as `client_kwargs` earlier

Reply:
I meant a variant of `configure_common_settings`, you need to configure track2 SDKs at some point. Something like:

```python
def configure_common_settings_track2(cli_ctx):
    client_kwargs = {}

    # FIXME change change_ssl_cert_verification to set client_kwargs['connection_verify']

    client_kwargs['logging_enable'] = True
    client_kwargs['user_agent'] = UA_AGENT
    try:
        client_kwargs['user_agent'] += os.environ[ENV_ADDITIONAL_USER_AGENT]
    except KeyError:
        pass
 
    # FIXME Add UA CliExtension

    client_kwargs['headers'] = dict(cli_ctx.data['headers'])  # Make a copy by using "dict"

    # FIXME finish this code with additional headers CommandName, etc.

    if 'x-ms-client-request-id' in cli_ctx.data['headers']:
        client_kwargs['request_id'] = cli_ctx.data['headers']['x-ms-client-request-id']
    return client_kwargs
```

Call this before you create the track2 client, and pass those kwargs. This is a change with track1, where you were creating the client before, and then setting "config"

Reply:
Got it. 

- One question is how can I know if the client is for track 2 or track 1? that means I need something like:

```
if track 2:
    configure_common_settings_track2()
client = client_type(cred, **client_kwargs)
if track 1:
    configure_common_settings_track1()
```
or do I just code as follows (always config for track 2),
```
configure_common_settings_track2()
client = client_type(cred, **client_kwargs)
if track 1:
    configure_common_settings_track1()
```

- Another question is that can all settings that are set in configure_common_settings_track1 be set in client_kwargs in track 2?

Reply:
Yes, all settings have an equivalent, otherwise let me know. 

And yes, something like that:
```python
if track2:
    client_kwargs.update(configure_common_settings_track2())
    client = client_type(credential=cred, **client_kwargs)
else: # track1
    client = client_type(credentials=cred, **client_kwargs)
    configure_common_settings_track1(client)
```